Core/GDCore/IDE/ProjectStripper.cpp

```diff
@@ -22,7 +24,28 @@ void GD_CORE_API ProjectStripper::StripProjectForExport(gd::Project& project) {
     project.GetLayout(i).GetEvents().Clear();
   }
 
-  project.ClearEventsFunctionsExtensions();
+  // Keep the EventsBasedObject object list because it's useful for the Runtime
+  // to create the child-object.
+  for (unsigned int extensionIndex = 0;
+       extensionIndex < project.GetEventsFunctionsExtensionsCount();
+       ++extensionIndex) {
+    auto &extension = project.GetEventsFunctionsExtension(extensionIndex);
+    auto &eventsBasedObjects = extension.GetEventsBasedObjects();
+    if (eventsBasedObjects.size() == 0) {
+      project.RemoveEventsFunctionsExtension(extension.GetName());
+      extensionIndex--;
```

Comment:
This is risky. If it happens for the first extension you would overflow (well, "underflow") the unsigned int from 0 to -1 which would become the highest unsigned int possible, which would then stop the loop.

Instead, remove the ++extensionIndex and do it only when not removing the extension.

Reply:
There should be no issue because the next thing to execute is the test and the increment of the for.

Reply:
> because the next thing to execute is the increment of the for.

Fair enough

> A while with an index is more risky to me because a misplaced closing bracket can lead to an infinite loop.

Sure, but I was advocating for a `for(unsigned int i = 0;i<xxxx;) { if(remove) {} else {++i;} }`.
At least I've been bitten in the past by unsigned int going from 0 to "max unsigned int" leading to infinite loop or skipping elements.